### PR TITLE
Cli fixes

### PR DESCRIPTION
**Pull request by jreidinger**
trello: https://trello.com/c/SHhhC8TH/1598-8-wrong-error-codes-in-cli

testing: manual

goal: fix all reported issues by customer.

### CODE REVIEW

File: src/clients/nfs_server.rb [hunk 1]
```diff
@@ -225,8 +228,10 @@ def check_packages
     # @param [Hash] options command options
     # @return whether successful
     def NfsServerStartHandler(options)
-      options = deep_copy(options)
-      return false if NfsServer.start
+      if NfsServer.start
+        CommandLine.Print(_("NFS server already running."))
+        return false
```
Comment (lslezak) [inline src/clients/nfs_server.rb]: BTW I'm not sure if starting NFS server is an error if it already is running... but as that was the previous implementation lets probably keep that...

File: src/clients/nfs_server.rb [hunk 2]
```diff
@@ -115,14 +116,16 @@ def main
             )
           },
           "enablev4"   => {
-            "type" => "string",
-            "help" => _(
-              "Yes/No option for enabling/disabling support for NFSv4."
+            "type"     => "enum",
+            "typespec" => ["yes", "no"],
+            "help"     => _(
+              "'yes'/'no option for enabling/disabling support for NFSv4."
```
Comment (lslezak) [inline src/clients/nfs_server.rb]: Missing quote

```suggestion
              "'yes'/'no' option for enabling/disabling support for NFSv4."
```

BTW do we support *translated* input, e.g. does it handle the Czech `ano`, `ne` as the input or only the English `yes`, `no` are expected?

If translations are not supported we should rather do something like

```ruby
_("%{yes}'/'%{no}' option for enabling/disabling support for NFSv4.") % { yes: "yes", no: "no" }
```

I'm afraid that the translators will also translate that "yes/no"...

File: src/clients/nfs_server.rb [hunk 3]
```diff
@@ -295,6 +303,8 @@ def NfsServerDeleteHandler(options)
         end
       end
 
+      CommandLine.Print(_("Mount point not found.")) if !deleted
```
Comment (lslezak) [inline src/clients/nfs_server.rb]: Maybe we could also print the mount point name so it's clear.

File: src/clients/nfs_server.rb (hunk 1 above)
Comment (jreidinger) [reply src/clients/nfs_server.rb]: yeah, I try to do minimal changes in behavior if someone already has working script. But in general I agree with you and probably main reason is to skip write operation.

File: src/clients/nfs_server.rb (hunk 2 above)
Comment (jreidinger) [reply src/clients/nfs_server.rb]: yes, that makes sense. We do not support translated input and to be honest it is hard to believe that someone really use that translated output in CLI which we target mainly for scripting.

File: src/clients/nfs_server.rb [hunk 4]
```diff
@@ -118,14 +118,22 @@ def main
           "enablev4"   => {
             "type"     => "enum",
             "typespec" => ["yes", "no"],
-            "help"     => _(
-              "'yes'/'no option for enabling/disabling support for NFSv4."
+            "help"     => format(
+              # TRANSLATORS %{yes} and %{no} is untranslated input values yes
```
Comment (lslezak) [inline src/clients/nfs_server.rb]:
```suggestion
              # TRANSLATORS %{yes} and %{no} are not translated input values "yes"
```

File: src/clients/nfs_server.rb [hunk 5]
```diff
@@ -118,14 +118,22 @@ def main
...
+              # and no which is not localized
```
Comment (lslezak) [inline src/clients/nfs_server.rb]:
```suggestion
              # and "no" which are not localized
```

File: src/clients/nfs_server.rb [hunk 6]
```diff
@@ -303,7 +311,10 @@ def NfsServerDeleteHandler(options)
         end
       end
 
-      CommandLine.Print(_("Mount point not found.")) if !deleted
+      if !deleted
+        # TRANSLATORS %s is directory where it should be mounted, but it is mounted there.
```
Comment (lslezak) [inline src/clients/nfs_server.rb]: Um, I do not get it. Should it be "but it is not found" or something like that?

File: src/clients/nfs_server.rb [hunk 7]
```diff
@@ -118,14 +118,22 @@ def main
...
+              _("%{yes}'/'%{no}' option for enabling/disabling support for NFSv4."),
+              yes: "yes", no: "no"
             )
           },
           "security"   => {
             "type"     => "enum",
             "typespec" => ["yes", "no"],
-            "help"     => _("'yes'/'no' option for enabling/disabling secure NFS.")
+            "help"     => format(
+              # TRANSLATORS %{yes} and %{no} is untranslated input values yes
```
Comment (lslezak) [inline src/clients/nfs_server.rb]: The same here.

File: src/clients/nfs_server.rb (hunk 6 above)
Comment (jreidinger) [reply src/clients/nfs_server.rb]: well, I do not want to confuse them that directory is not found. In fact mount point is not found in nfs server configuration.

**Pull request merged**